### ISSUE TITLE
Error building pose tracking framework for iOS 

### ISSUE THREAD

**Issue by V-m1r** [google/mediapipe#3173]
Hi, I'm trying to build a pose tracking framework for iOS.
Mediapipe version is 0.8.9
Bazel version -  4.2.1

In posetrackinggpu/BUILD I have:
```
load(
    "@build_bazel_rules_apple//apple:ios.bzl",
    "ios_framework",
)

ios_framework(
    name = "PoseTrackingGpuApp",

    hdrs = [
        "PoseTrackingViewController.h",
    ],
    
    infoplists = ["Info.plist"],
    bundle_id = "com.PoseTracker.vmir",
    families = ["iphone", "ipad"],
    minimum_os_version = "10.0",
    deps = [
        ":PoseTrackingGpuAppLibrary",
        "@ios_opencv//:OpencvFramework",
    ],
)

objc_library(
    name = "PoseTrackingGpuAppLibrary",
    srcs = [
        "PoseTrackingViewController.mm",
    ],
    hdrs = [
        "PoseTrackingViewController.h",
    ],
    
    sdk_frameworks = [
        "AVFoundation",
        "CoreGraphics",
        "CoreMedia",
        "UIKit",
    ],
    
    copts = ["-std=c++17"],
    
    data = [ 
        "//mediapipe/graphs/pose_tracking:pose_tracking_gpu.binarypb",
        "//mediapipe/modules/pose_detection:pose_detection.tflite",
        "//mediapipe/modules/pose_landmark:pose_landmark_full.tflite",
    ],
    
    deps = [
        "//mediapipe/examples/ios/common:CommonMediaPipeAppLibrary",
            "//mediapipe/objc:mediapipe_framework_ios",
        "//mediapipe/objc:mediapipe_input_sources_ios",
        "//mediapipe/objc:mediapipe_layer_renderer",
    ] + select({
        "//mediapipe:ios_i386": [],
        "//mediapipe:ios_x86_64": [],
        "//conditions:default": [
            "//mediapipe/graphs/pose_tracking:pose_tracking_gpu_deps",
            "//mediapipe/framework/formats:landmark_cc_proto",
        ],
    }),
)
```

I use this command to build the framework:
`
bazel build --config=ios_arm64 mediapipe/examples/ios/posetrackinggpu:PoseTrackingGpuApp --verbose_failures
`
And get the following error:
```
ERROR: /private/var/tmp/_bazel_vladimirshetnikov/938e5bf8c35858e1e32eb682a6b7501b/external/org_tensorflow/tensorflow/core/util/BUILD:379:24: Action external/org_tensorflow/tensorflow/core/util/version_info.cc failed: (Exit 127): bash failed: error executing command 
  (cd /private/var/tmp/_bazel_vladimirshetnikov/938e5bf8c35858e1e32eb682a6b7501b/sandbox/darwin-sandbox/699/execroot/mediapipe && \
  exec env - \
    PATH=/opt/homebrew/bin:/opt/homebrew/sbin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Library/Apple/usr/bin \
  /bin/bash -c 'bazel-out/ios-arm64-min10.0-applebin_ios-ios_arm64-fastbuild-ST-d39cb6bc319b/bin/external/org_tensorflow/tensorflow/tools/git/gen_git_source --generate "$@" --git_tag_override=${GIT_TAG_OVERRIDE:-}' '' external/local_config_git/gen/spec.json external/local_config_git/gen/head external/local_config_git/gen/branch_ref bazel-out/ios-arm64-min10.0-applebin_ios-ios_arm64-fastbuild-ST-50603ff4af65/bin/external/org_tensorflow/tensorflow/core/util/version_info.cc)
Execution platform: @local_execution_config_platform//:platform

Use --sandbox_debug to see verbose messages from the sandbox
env: python: No such file or directory
Target //mediapipe/examples/ios/posetrackinggpu:PoseTrackingGpuApp failed to build
INFO: Elapsed time: 49.864s, Critical Path: 30.02s
INFO: 838 processes: 144 internal, 694 darwin-sandbox.
FAILED: Build did NOT complete successfully

```

What could be the issue and how to solve it?

Thank you!

Edit:

When I try to run the initial BUILD file I also get an error:

```
ERROR: /Users/vladimirshetnikov/Desktop/mediapipe/mediapipe/examples/ios/posetrackinggpu/BUILD:6:14: StoryboardCompile mediapipe/examples/ios/posetrackinggpu/PoseTrackingGpuApp-intermediates/Base.lproj_storyboardc/LaunchScreen.storyboardc failed: (Exit 127): xctoolrunner failed: error executing command 
  (cd /private/var/tmp/_bazel_vladimirshetnikov/938e5bf8c35858e1e32eb682a6b7501b/execroot/mediapipe && \
  exec env - \
    APPLE_SDK_PLATFORM=iPhoneOS \
    APPLE_SDK_VERSION_OVERRIDE=15.4 \
    XCODE_VERSION_OVERRIDE=13.3.0.13E113 \
  bazel-out/host/bin/external/build_bazel_rules_apple/tools/xctoolrunner/xctoolrunner ibtool --compilation-directory '[ABSOLUTE]bazel-out/applebin_ios-ios_arm64-fastbuild-ST-50603ff4af65/bin/mediapipe/examples/ios/posetrackinggpu/PoseTrackingGpuApp-intermediates/Base.lproj_storyboardc' --minimum-deployment-target 10.0 --target-device iphone --target-device ipad --module PoseTrackingGpuApp '[ABSOLUTE]mediapipe/examples/ios/common/Base.lproj/LaunchScreen.storyboard')
Execution platform: @local_execution_config_platform//:platform
env: python: No such file or directory
Target //mediapipe/examples/ios/posetrackinggpu:PoseTrackingGpuApp failed to build
INFO: Elapsed time: 82.713s, Critical Path: 45.61s
INFO: 2138 processes: 91 internal, 2047 darwin-sandbox.
FAILED: Build did NOT complete successfully

```

**Comment by sureshdagooglecom** [comment]
Hi @V-m1r ,
Could you provide steps to reproduce this issue.

**Comment by V-m1r** [comment]
Hello @sureshdagooglecom,
Thank you for your comment!
So I cloned latest madiapipe,
changed the BUILD file contents In posetrackinggpu to
```
 load(
    "@build_bazel_rules_apple//apple:ios.bzl",
    "ios_framework",
)

ios_framework(
    name = "PoseTrackingGpuApp",

    hdrs = [
        "PoseTrackingViewController.h",
    ],
    
    infoplists = ["Info.plist"],
    bundle_id = "com.PoseTracker.vmir",
    families = ["iphone", "ipad"],
    minimum_os_version = "10.0",
    deps = [
        ":PoseTrackingGpuAppLibrary",
        "@ios_opencv//:OpencvFramework",
    ],
)

objc_library(
    name = "PoseTrackingGpuAppLibrary",
    srcs = [
        "PoseTrackingViewController.mm",
    ],
    hdrs = [
        "PoseTrackingViewController.h",
    ],
    
    sdk_frameworks = [
        "AVFoundation",
        "CoreGraphics",
        "CoreMedia",
        "UIKit",
    ],
    
    copts = ["-std=c++17"],
    
    data = [ 
        "//mediapipe/graphs/pose_tracking:pose_tracking_gpu.binarypb",
        "//mediapipe/modules/pose_detection:pose_detection.tflite",
        "//mediapipe/modules/pose_landmark:pose_landmark_full.tflite",
    ],
    
    deps = [
        "//mediapipe/examples/ios/common:CommonMediaPipeAppLibrary",
            "//mediapipe/objc:mediapipe_framework_ios",
        "//mediapipe/objc:mediapipe_input_sources_ios",
        "//mediapipe/objc:mediapipe_layer_renderer",
    ] + select({
        "//mediapipe:ios_i386": [],
        "//mediapipe:ios_x86_64": [],
        "//conditions:default": [
            "//mediapipe/graphs/pose_tracking:pose_tracking_gpu_deps",
            "//mediapipe/framework/formats:landmark_cc_proto",
        ],
    }),
)
```
ran following command:
`bazel build --config=ios_arm64 mediapipe/examples/ios/posetrackinggpu:PoseTrackingGpuApp --verbose_failures`

Then bazel started. It compiled some stuff and was close to finish, but then aborted and gave the following error:
```
ERROR: /private/var/tmp/_bazel_vladimirshetnikov/938e5bf8c35858e1e32eb682a6b7501b/external/org_tensorflow/tensorflow/core/util/BUILD:379:24: Action external/org_tensorflow/tensorflow/core/util/version_info.cc failed: (Exit 127): bash failed: error executing command 
  (cd /private/var/tmp/_bazel_vladimirshetnikov/938e5bf8c35858e1e32eb682a6b7501b/sandbox/darwin-sandbox/1/execroot/mediapipe && \
  exec env - \
    PATH=/opt/homebrew/bin:/opt/homebrew/sbin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Library/Apple/usr/bin \
  /bin/bash -c 'bazel-out/ios-arm64-min10.0-applebin_ios-ios_arm64-fastbuild-ST-d39cb6bc319b/bin/external/org_tensorflow/tensorflow/tools/git/gen_git_source --generate "$@" --git_tag_override=${GIT_TAG_OVERRIDE:-}' '' external/local_config_git/gen/spec.json external/local_config_git/gen/head external/local_config_git/gen/branch_ref bazel-out/ios-arm64-min10.0-applebin_ios-ios_arm64-fastbuild-ST-50603ff4af65/bin/external/org_tensorflow/tensorflow/core/util/version_info.cc)
Execution platform: @local_execution_config_platform//:platform

Use --sandbox_debug to see verbose messages from the sandbox
env: python: No such file or directory
Target //mediapipe/examples/ios/posetrackinggpu:PoseTrackingGpuApp failed to build
INFO: Elapsed time: 17.108s, Critical Path: 1.01s
INFO: 81 processes: 81 internal.
FAILED: Build did NOT complete successfully
```

My full terminal log:
```
vladimirshetnikov@Vladimirs-MacBook-Pro ~ % cd desktop
vladimirshetnikov@Vladimirs-MacBook-Pro desktop % cd mediapipe
vladimirshetnikov@Vladimirs-MacBook-Pro mediapipe % bazel build --config=ios_arm64 mediapipe/examples/ios/posetrackinggpu:PoseTrackingGpuApp --verbose_failures
Starting local Bazel server and connecting to it...
DEBUG: /private/var/tmp/_bazel_vladimirshetnikov/938e5bf8c35858e1e32eb682a6b7501b/external/org_tensorflow/third_party/repo.bzl:122:14: 
Warning: skipping import of repository 'com_google_absl' because it already exists.
DEBUG: /private/var/tmp/_bazel_vladimirshetnikov/938e5bf8c35858e1e32eb682a6b7501b/external/org_tensorflow/third_party/repo.bzl:122:14: 
Warning: skipping import of repository 'com_google_benchmark' because it already exists.
DEBUG: /private/var/tmp/_bazel_vladimirshetnikov/938e5bf8c35858e1e32eb682a6b7501b/external/org_tensorflow/third_party/repo.bzl:122:14: 
Warning: skipping import of repository 'pybind11_bazel' because it already exists.
DEBUG: /private/var/tmp/_bazel_vladimirshetnikov/938e5bf8c35858e1e32eb682a6b7501b/external/org_tensorflow/third_party/repo.bzl:122:14: 
Warning: skipping import of repository 'com_google_protobuf' because it already exists.
DEBUG: /private/var/tmp/_bazel_vladimirshetnikov/938e5bf8c35858e1e32eb682a6b7501b/external/org_tensorflow/third_party/repo.bzl:122:14: 
Warning: skipping import of repository 'com_google_googletest' because it already exists.
DEBUG: /private/var/tmp/_bazel_vladimirshetnikov/938e5bf8c35858e1e32eb682a6b7501b/external/org_tensorflow/third_party/repo.bzl:122:14: 
Warning: skipping import of repository 'com_github_gflags_gflags' because it already exists.
DEBUG: /private/var/tmp/_bazel_vladimirshetnikov/938e5bf8c35858e1e32eb682a6b7501b/external/org_tensorflow/third_party/repo.bzl:122:14: 
Warning: skipping import of repository 'build_bazel_rules_apple' because it already exists.
DEBUG: /private/var/tmp/_bazel_vladimirshetnikov/938e5bf8c35858e1e32eb682a6b7501b/external/org_tensorflow/third_party/repo.bzl:122:14: 
Warning: skipping import of repository 'build_bazel_rules_swift' because it already exists.
DEBUG: /private/var/tmp/_bazel_vladimirshetnikov/938e5bf8c35858e1e32eb682a6b7501b/external/org_tensorflow/third_party/repo.bzl:122:14: 
Warning: skipping import of repository 'build_bazel_apple_support' because it already exists.
DEBUG: /private/var/tmp/_bazel_vladimirshetnikov/938e5bf8c35858e1e32eb682a6b7501b/external/org_tensorflow/third_party/repo.bzl:122:14: 
Warning: skipping import of repository 'xctestrunner' because it already exists.
DEBUG: /private/var/tmp/_bazel_vladimirshetnikov/938e5bf8c35858e1e32eb682a6b7501b/external/org_tensorflow/third_party/repo.bzl:122:14: 
Warning: skipping import of repository 'pybind11' because it already exists.
WARNING: /Users/vladimirshetnikov/Desktop/mediapipe/mediapipe/framework/tool/BUILD:182:24: in cc_library rule //mediapipe/framework/tool:field_data_cc_proto: target '//mediapipe/framework/tool:field_data_cc_proto' depends on deprecated target '@com_google_protobuf//:cc_wkt_protos': Only for backward compatibility. Do not use.
WARNING: /Users/vladimirshetnikov/Desktop/mediapipe/mediapipe/framework/BUILD:54:24: in cc_library rule //mediapipe/framework:calculator_cc_proto: target '//mediapipe/framework:calculator_cc_proto' depends on deprecated target '@com_google_protobuf//:cc_wkt_protos': Only for backward compatibility. Do not use.
INFO: Analyzed target //mediapipe/examples/ios/posetrackinggpu:PoseTrackingGpuApp (171 packages loaded, 7473 targets configured).
INFO: Found 1 target...
ERROR: /private/var/tmp/_bazel_vladimirshetnikov/938e5bf8c35858e1e32eb682a6b7501b/external/org_tensorflow/tensorflow/core/util/BUILD:379:24: Action external/org_tensorflow/tensorflow/core/util/version_info.cc failed: (Exit 127): bash failed: error executing command 
  (cd /private/var/tmp/_bazel_vladimirshetnikov/938e5bf8c35858e1e32eb682a6b7501b/sandbox/darwin-sandbox/1/execroot/mediapipe && \
  exec env - \
    PATH=/opt/homebrew/bin:/opt/homebrew/sbin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Library/Apple/usr/bin \
  /bin/bash -c 'bazel-out/ios-arm64-min10.0-applebin_ios-ios_arm64-fastbuild-ST-d39cb6bc319b/bin/external/org_tensorflow/tensorflow/tools/git/gen_git_source --generate "$@" --git_tag_override=${GIT_TAG_OVERRIDE:-}' '' external/local_config_git/gen/spec.json external/local_config_git/gen/head external/local_config_git/gen/branch_ref bazel-out/ios-arm64-min10.0-applebin_ios-ios_arm64-fastbuild-ST-50603ff4af65/bin/external/org_tensorflow/tensorflow/core/util/version_info.cc)
Execution platform: @local_execution_config_platform//:platform

Use --sandbox_debug to see verbose messages from the sandbox
env: python: No such file or directory
Target //mediapipe/examples/ios/posetrackinggpu:PoseTrackingGpuApp failed to build
INFO: Elapsed time: 17.108s, Critical Path: 1.01s
INFO: 81 processes: 81 internal.
FAILED: Build did NOT complete successfully

```

So the issue is in failed bash. It cannot execute this: 

```
/private/var/tmp/_bazel_vladimirshetnikov/938e5bf8c35858e1e32eb682a6b7501b/external/org_tensorflow/tensorflow/core/util/BUILD:379:24: Action external/org_tensorflow/tensorflow/core/util/version_info.cc failed: (Exit 127): bash failed: error executing command 
  (cd /private/var/tmp/_bazel_vladimirshetnikov/938e5bf8c35858e1e32eb682a6b7501b/sandbox/darwin-sandbox/1/execroot/mediapipe && \
  exec env - \
    PATH=/opt/homebrew/bin:/opt/homebrew/sbin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Library/Apple/usr/bin \
  /bin/bash -c 'bazel-out/ios-arm64-min10.0-applebin_ios-ios_arm64-fastbuild-ST-d39cb6bc319b/bin/external/org_tensorflow/tensorflow/tools/git/gen_git_source --generate "$@" --git_tag_override=${GIT_TAG_OVERRIDE:-}' '' external/local_config_git/gen/spec.json external/local_config_git/gen/head external/local_config_git/gen/branch_ref bazel-out/ios-arm64-min10.0-applebin_ios-ios_arm64-fastbuild-ST-50603ff4af65/bin/external/org_tensorflow/tensorflow/core/util/version_info.cc)

```

**Comment by sureshdagooglecom** [comment]
Could you please check this two issues where community already tried doing the same https://github.com/google/mediapipe/issues/115, https://github.com/google/mediapipe/issues/364. If you still not able to find the resolution please reach us back.

**Comment by V-m1r** [comment]
Issue got solved with an updated Mediapipe version (0.8.9). I have successively built the framework and integrated it into app. Thanks

**Comment by Pogosito** [comment]
@V-m1r 
Thank you very much for your build file, with the help of I also managed to build a framework

But I'm having trouble linking it

I added generated PoseTrackingGpuApp.framework in my app. But I have error when try build project. 

Please tell me how you managed to link the framework 

![Снимок экрана 2022-08-24 в 23 31 30](https://user-images.githubusercontent.com/56889876/186517543-b27d92f4-1d2a-411a-8c46-8b3201a3608a.png)
![Снимок экрана 2022-08-24 в 23 32 34](https://user-images.githubusercontent.com/56889876/186517739-b7802f99-5cf5-4145-ac26-79696aca5589.png)
![Снимок экрана 2022-08-24 в 23 35 32](https://user-images.githubusercontent.com/56889876/186518281-dd4d53c0-2c72-4d09-beaf-9f0b1e5e07b1.png)
